src/repr/src/adt/datetime.rs

```diff
@@ -873,13 +873,23 @@ impl ParsedDateTime {
     /// # Arguments
     ///
     /// * `value` is a SQL-formatted TIMESTAMP string.
-    pub fn build_parsed_datetime_timestamp(value: &str) -> Result<ParsedDateTime, String> {
+    pub fn build_parsed_datetime_timestamp(
+        value: &str,
+        was_bc: bool,
+    ) -> Result<ParsedDateTime, String> {
         let mut pdt = ParsedDateTime::default();
 
         let mut ts_actual = tokenize_time_str(value)?;
 
         fill_pdt_date(&mut pdt, &mut ts_actual)?;
 
+        if was_bc {
+            pdt.year = pdt.year.map(|mut y| {
+                y.unit = y.unit * -1;
```

Comment:
I was surprised with the multiplication. Is this doing something different than `y.unit = -y.unit`?

Reply:
it's not I will refactor tot that. I think it's the same thing? but no idea what the eventuall llvm would be in either case.

Reply:
done